assets/src/apps/delivery/store/features/adaptivity/actions/triggerCheck.ts

```diff
@@ -91,7 +90,22 @@ export const triggerCheck = createAsyncThunk(
         }
       });
     });
-    const stateSnapshot = { ...allResponseState, ...modifiedExtrinsicState };
+
+    const spanshot = getEnvState(defaultGlobalEnv);
+    const getGlobalSnapshot = [spanshot].reduce((collect: any, variableSnappshot: any) => {
+      Object.keys(spanshot).forEach((key) => {
+        if (key.indexOf('app.') === 0 || key.indexOf('variables.') === 0) {
+          collect[key] = spanshot[key];
+        }
+      });
+      return collect;
+    }, {});
+    const updatedExtrinsicState = { ...modifiedExtrinsicState, ...getGlobalSnapshot };
+    await dispatch(setExtrinsicState({ state: updatedExtrinsicState }));
```

Comment:
I don't think that we want to write variables and app into Extrinsic state, just send the values into the "check" call. the variables are one shot properties of the lesson model, and the app ones are from user state. when we implement user state storage then we can collect the values from that (wherever they are stored), but still we should just include them in the call to check, not write them to the lesson attempt